internal/watermill/middleware/dlq.go

```diff
@@ -0,0 +1,96 @@
+package middleware
+
+import (
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/openmeterio/openmeter/internal/watermill"
+)
+
+const (
+	DLQErrorReasonKey   = "reason_dlq"
+	DLQOriginalTopicKey = "original_topic"
+)
+
+// DLQMsgProcessorFunc determines whether any messages should be published to the DLQ based on the original message and the error.
+// Even if DLQMsgProcessorFunc returns a non-nil slice of messages, if error is nil no messages will be published to the DLQ.
+type DLQMsgProcessorFunc func(original *message.Message, err error) []*message.Message
+
+// defaultProcessor listens for RetryableErrors and publishes the messages contained in that
+var defaultProcessor DLQMsgProcessorFunc = func(original *message.Message, err error) []*message.Message {
+	if retryable, ok := err.(watermill.RetryableError); ok {
```

Comment:
Please use the errorsx.ErrorAs (or errors.As), this will not work properly if any wrapping occurs.

Reply:
on a more philosophical note, if we rely on the error data/methods for our behavior we should probably not include wrapped errors, but I don't realistically see it turning into an issue